### PR TITLE
Refactored with async/await

### PR DESCRIPTION
**Pull request by Zearin**
This is a continuation of Issue #303 as a Pull Request (for easier code review, line comments, etc.).

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -17,6 +16,13 @@ const util = require('util')
  */
 const rm      = util.promisify(require('rimraf'))
 
+
+/**
+ * Helpers
+ */
+const absolute = (s) => {return path.resolve(s) === s}
```
Comment (Ajedi32) [inline lib/index.js]: I looked at the code for `absolute`. It's a polyfill for [`path.isAbsolute`](https://nodejs.org/api/path.html#path_path_isabsolute_path), which was added in node 0.12.

Given that we don't support versions of node older than that anymore, you can shorten this line further to `const absolute = path.isAbsolute.bind(path)` or just get rid of it entirely and replace the calls further down with calls to `path.isAbsolute`.

File: package.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
     "precoverage": "npm install",
     "coverage": "node $HARMONY_OPTS ./node_modules/.bin/istanbul cover ./node_modules/.bin/_mocha",
     "precoveralls": "npm run coverage",
-    "coveralls": "coveralls < coverage/lcov.info"
+    "coveralls": "coveralls < coverage/lcov.info",
+    "lint": "eslint index.js lib/*.js bin/*"
```
Comment (Ajedi32) [inline package.json]: Probably want to run this on Travis so it's actually enforced for new PRs.

File: .travis.yml [hunk 3]
```diff
@@ -2,8 +2,6 @@ language: node_js
 sudo: false
 
 node_js:
-  - "6"
-  - "7"
```
Comment (Ajedi32) [inline .travis.yml]: Is there any way we can polyfill promisify instead? Node 6 is an LTS release and still supported by Node for another year: https://github.com/nodejs/Release#release-schedule

**Comment by Ajedi32** [comment]
FYI, there are now conflicts that need to be resolved, since a lot of the changes in this PR were duplicated in #306.

**Comment by Zearin** [comment]
> FYI, there are now conflicts that need to be resolved, since a lot of the changes in this PR were duplicated in #306.

No problem.  Since you wanted a few of the non-refactor changes in separate PRs, I’ve got a few more of them to complete first.  So, I’ll close this PR for now, and reopen a new one after all my smaller PRs are merged.

**Comment by fhemberger** [comment]
@Ajedi32 @Zearin Hi, any updates on async/await support for Metalsmith?

**Comment by Ajedi32** [comment]
There hasn't been much progress on this lately since I've been spending most of my time on other projects. Maybe I'll find some time later, but for now it's kinda on hold unless someone else wants to champion it.

**Comment by webketje** [comment]
@fhemberger in case you're still interested; upcoming release metalsmith 2.5 will support promise-based flows for all methods

**Pull request closed without merging**